src/Weeds3dViewer/Weeds3dViewer.js

```diff
@@ -0,0 +1,228 @@
+  await axios({
+    method: 'GET',
+    url: onfarmAPI + '/raw?output=json&table=site_information',
+    headers: {
+      'x-api-key': apikey,
+    },
+    responseType: 'json',
+    timeout: 5000,
+  })
+    .then((res) => {
+      const data = res.data;
+      Object.keys(data).forEach((key) => {
+        listOfCodes.push(data[key].code);
+      });
+    })
+    .catch((err) => {
+      console.log('API error: ' + err);
+    });
+
+  return listOfCodes;
+};
+
+const fetchAPIData = async (codes, getTokenSilently, setVideos, setLoading) => {
+  const data = { codes: codes };
+  await callAzureFunction(data, '/weeds3d/videos', 'POST', getTokenSilently)
+    .then((res) => {
+      setVideos(res.jsonResponse.files);
+      setLoading(false);
+    })
+    .catch((err) => {
+      console.log('API error: ' + err);
+    });
+};
+
+const Weeds3dViewer = () => {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const userInfo = useSelector((state) => state.userInfo);
+  const { getTokenSilently } = useAuth0();
+  const { user } = useAuth0();
+  const height = window.innerHeight;
+  const [snackbarData, setSnackbarData] = useState({
+    open: false,
+    text: '',
+    severity: 'success',
+  });
+
+  useEffect(() => {
+    const fetchData = async (apikey) => {
+      const codes = await fetchCodes(apikey);
+      await fetchAPIData(codes, getTokenSilently, setVideos, setLoading);
+    };
+
+    if (!userInfo.apikey) return false;
+    fetchData(userInfo.apikey);
+  }, [getTokenSilently, userInfo.apikey]);
+
+  return (
+    <Grid container>
+      {loading ? (
+        <CustomLoader />
+      ) : (
+        <Grid item lg={12}>
+          <Snackbar
+            anchorOrigin={{
+              vertical: 'bottom',
+              horizontal: 'center',
+            }}
+            open={snackbarData.open}
+            autoHideDuration={10000}
+            onClose={() => setSnackbarData({ ...snackbarData, open: !snackbarData.open })}
+          >
+            <Alert severity={snackbarData.severity}>{snackbarData.text}</Alert>
+          </Snackbar>
+          <MaterialTable
+            columns={tableHeaderOptions}
+            data={videos}
+            title={
+              <div>
+                <div>
+                  <Typography
+                    variant={'h6'}
+                    style={{
+                      whiteSpace: 'nowrap',
+                      overflow: 'hidden',
+                      textOverflow: 'ellipsis',
+                    }}
+                  >
+                    {'Weeds 3d Videos'}
+                  </Typography>
+                </div>
+              </div>
+            }
+            options={{
+              defaultExpanded: false,
+              padding: 'default',
+              exportButton: false,
+              exportFileName: 'Weeds 3d Videos',
+              addRowPosition: 'last',
+              exportAllData: false,
+              pageSize: videos.length,
```

Comment:
this is only showing one row per page when i build it. it should show all of the rows